lib/ExecutionContext.js

```diff
@@ -94,8 +94,23 @@ class ExecutionContext {
     if (typeof pageFunction !== 'function')
       throw new Error('The following is not a function: ' + pageFunction);
 
+    let functionText = pageFunction.toString();
+    try {
+      eval(functionText);
```

Comment:
I'd be worried about unintended side effects here. Consider: `function functionText(){return 5}`.

Reply:
good catch; switched to `new Function` instead